### PR TITLE
w3c compliance

### PR DESCRIPTION
**Pull request by dpgraham**
* Update 'createSession' to comply with W3C
* Updated tests to comply with W3C
* Wrote tests that check the response for W3C createSession and MJSONWP createSession to verify that they follow the standard

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ class FakeDriver extends BaseDriver {
     };
   }
 
-  async createSession (caps, reqCaps, otherSessionData = []) {
+  async createSession (...args) {
+    let otherSessionData = args[3] || [];
```
Comment (mykola-mokhnach) [inline lib/driver.js]:
```
let [caps, desiiredCaps, otherSessionData, ...] = args;
otherSessionData = otherSessionData || []
```

might be safer

File: test/driver-e2e-specs.js [hunk 2]
```diff
@@ -53,4 +54,34 @@ describe('FakeDriver - via HTTP', () => {
...
+        }
+      });
+      res.value.capabilities.should.deep.equal(Object.assign({}, DEFAULT_CAPS, {
+        fakeCap: 'Foo',
+      }));
+      res.value.sessionId.should.exist;
+      should.not.exist(res.status);
+      res = await request.delete(`http://${TEST_HOST}:${TEST_PORT}/wd/hub/session/${res.value.sessionId}`);
+
+      // Now create a MJSONWP session and check the response
```
Comment (mykola-mokhnach) [inline test/driver-e2e-specs.js]: I'd rather create two separate tests

File: package.json [hunk 3]
```diff
@@ -25,13 +25,14 @@
...
-    "appium-support": "^2.5.0",
+    "appium-base-driver": "^2.18.4",
+    "appium-support": "^2.11.1",
     "asyncbox": "^2.0.2",
-    "babel-runtime": "=5.8.24",
-    "bluebird": "^3.1.1",
+    "babel-core": "5.8.24",
+    "babel-runtime": "5.8.24",
+    "bluebird": "^3.5.1",
     "lodash": "^3.9.1",
```
Comment (mykola-mokhnach) [inline package.json]: can we also update lodash to 4+ like in other modules?

Comment (imurchie) [reply package.json]: There are number of places where we could update. https://github.com/appium/appium/issues/9866

**Comment by dpgraham** [comment]
Fixes made. Will do the lodash updates later as a separate PR for a bunch of different repos.

**Comment by mykola-mokhnach** [comment]
I assume jshint verification has to be removed from Travis and replaced with eslint, otherwise the stuff will always fail: https://travis-ci.org/appium/appium-fake-driver/jobs/322337176

Ping @imurchie

**Comment by imurchie** [comment]
See #15

**Pull request merged**